framework/sandstone.cpp

```diff
@@ -2120,7 +2118,12 @@ TestResult run_one_test(int *tc, const struct test *test, SandstoneApplication::
     /* First we go to do our -- possibly fractured -- normal run, we'll do retries after */
     for (sApp->current_iteration_count = 0;; ++sApp->current_iteration_count) {
         init_internal(test);
+
+        // calculate starttime->endtime, reduce the overhead to have better test runtime calculations
+        sApp->current_test_endtime = calculate_wallclock_deadline(sApp->current_test_duration - runtime, &sApp->current_test_starttime);
         state = run_one_test_once(tc, test);
+        runtime += MonotonicTimePoint::clock::now() - sApp->current_test_starttime;
```

Comment:
internal init/cleanup aren't included in the test runtime, but overall runtime includes all the overhead stuff.

Reply:
Fair.